response_operations_ui/controllers/party_controller.py

```diff
@@ -0,0 +1,125 @@
+import logging
+
+import requests
+from requests.exceptions import HTTPError, RequestException
+from structlog import wrap_logger
+
+from response_operations_ui import app
+from response_operations_ui.controllers.survey_controllers import get_survey_by_id
+from response_operations_ui.exceptions.exceptions import ApiError, UpdateContactDetailsException
+from response_operations_ui.forms import EditContactDetailsForm
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+def get_respondent_by_party_id(party_id):
+    logger.debug("Get respondent details")
+    url = f'{app.config["PARTY_URL"]}/party-api/v1/respondents/id/{party_id}'
+    response = requests.get(url=url, auth=app.config['PARTY_AUTH'])
+
+    try:
+        response.raise_for_status()
+    except (HTTPError, RequestException):
+        logger.exception("Respondent retrieval failed", party_id=party_id)
```

Comment:
The get respondent api can respond with a 400 or 404 which we don't want to log at error level (which is the log level for logger.exception)
```
log_level = logger.warning if response.status_code in (400, 404) else logger.exception
log_level("Respondent retrieval failed", party_id=party_id)
```